lnwallet/wallet.go

```diff
@@ -805,18 +805,29 @@ func (l *LightningWallet) initOurContribution(reservation *ChannelReservation,
 	if err != nil {
 		return err
 	}
-	revocationRoot, err := l.DerivePrivKey(nextRevocationKeyDesc)
+
+	if nextRevocationKeyDesc.Index == 0 {
```

Comment:
Why is this needed? Should at least have a comment.

Reply:
DerivePrivKey in keychain/btcwallet.go special cases Index == 0 implicitly

Reply:
Added a comment